### DISCUSSION TITLE
前端项目发布新版本后提示用户刷新页面更新

### DISCUSSION THREAD

**Discussion by pangxzi** [umijs/umi#9457] [Q&A]
https://blog.csdn.net/guzhao593/article/details/93893787
类似这样在umi中怎么实现？

**Comment by fz6m** [comment]
手动提示刷新多见于 PWA 应用，对于大部分 umi 的 SPA 应用来说，使用 [hash](https://umijs.org/docs/api/config#hash) 选项，构建出包含 hash 的产物，在 网关 / cdn / nginx 给 js 设定一个较长期限的缓存，给 index.html 设定一个很短的缓存、或不设缓存即可。

**Reply by pangxzi** [reply]
现在就是没有缓存，但是修改代码更新后如果没有刷新浏览器就会找不到之前的一些js文件
![image](https://user-images.githubusercontent.com/52907799/192916472-38ebf4cd-1de6-43cb-9a83-a31d06d2e394.png)

**Reply by lzw10168** [reply]
我也碰到这个问题了. 请问一下有解决办法了吗 .  或者捕获错误强制刷新.

**Reply by fz6m** [reply]
请排查公司的 Nginx 对 `.html` 文件下发的缓存配置，设置为 `no-cache` 无需缓存。
